libp2p/NodeTable.h

```diff
@@ -177,26 +177,26 @@ class NodeTable : UDPSocketEvents
 protected:
     /// Constants for Kademlia, derived from address space.
 
-    static unsigned const s_addressByteSize = h256::size;					///< Size of address type in bytes.
-    static unsigned const s_bits = 8 * s_addressByteSize;					///< Denoted by n in [Kademlia].
-    static unsigned const s_bins = s_bits - 1;  ///< Size of m_buckets (excludes root, which is us).
-    static unsigned const s_maxSteps = boost::static_log2<s_bits>::value;	///< Max iterations of discovery. (discover)
+    static constexpr unsigned s_addressByteSize = h256::size;					///< Size of address type in bytes.
```

Comment:
Unfortunately, even when they are `constexpr` now you still have to allocate memory for them. This is some kind of C++11 bug, fixed in newer revisions.

Easy way to workaround this, is to move them out of class to .cpp file. This also saves us including `boost::static_log2` in the header file.

Reply:
There's `m_buckets` member definition that depends on `s_bins`, so I can't just move it...

Reply:
I've added definitions to cpp for the ones clang was complaining about